Archive.cs

```diff
@@ -30,22 +31,22 @@ public static void Unpack(Stream ArchiveRaw, string Folder)
                 if (!Psarc.FileNames.ContainsKey(filenameHash))
                 {
                     Console.WriteLine("Archive Contains a hash which is not in filenames table: " + filenameHash);
-                    FileName = "_Unknowns\\" + filenameHash.Replace("-", "") + ".bin";
+                    FileName = "_Unknowns" + Path.DirectorySeparatorChar.ToString() + filenameHash.Replace("-", "") + ".bin";
                 }
                 else
                 {
-                    FileName = Psarc.FileNames[filenameHash].Replace("/", "\\");
+                    FileName = Psarc.FileNames[filenameHash].Replace("/", Path.DirectorySeparatorChar.ToString());
                 }
 
 
-                if (FileName.StartsWith("\\"))
+                if (FileName.StartsWith(Path.DirectorySeparatorChar.ToString()))
                     FileName = FileName.Remove(0, 1);
 
                 try
                 {
                     TryUnpack(Psarc.Reader, out HugeMemoryStream FileWriter, Psarc.Entries[i], Psarc.ZSizes, Psarc.BlockSize, Psarc.CompressionType);
-                    IOHelper.CheckFolderExists(FileName, Folder);
-                    Stream fileHandle = File.Open(@"\\?\" + Path.Combine(Folder, FileName), FileMode.Create, FileAccess.Write);
```

Comment:
The `\\?\` must be present because of the MAX_PATH variable in Windows (it is 256 characters and if the path is more than 256 characters the tool will crash, and `\\?\` will bypass that limit)